.circleci/config.yml

```diff
@@ -19,16 +19,21 @@ anchors:
 
   java_defaults: &java_defaults
     <<: *defaults
+    resource_class: medium+
     environment:
       # As best I can tell (dmohs, 7 Feb '17), this is the only way to set a memory limit that Java
       # processes executed within CircleCI's docker containers will respect. Very helpful resource:
       # https://circleci.com/blog/how-to-handle-java-oom-errors/
       #
       # In Feb 2017, this was set to 3G. But in Feb 2019 (RW-2194) we started seeing OOM errors,
       # so we bumped this down further to 2G.
-      JAVA_TOOL_OPTIONS: -Xmx2g
+      # JAVA_TOOL_OPTIONS: -Xmx2g
+
+      # Configure the JVM and Gradle: give it a bigger memory to avoid OOM errors
+      # https://circleci.com/docs/2.0/java-oom/#using-java-environment-variables-to-set-memory-limits
+      _JAVA_OPTIONS: "-Xmx3g"
       # https://docs.gradle.org/6.3/userguide/gradle_daemon.html#sec:disabling_the_daemon
-      GRADLE_OPTS: "-Dorg.gradle.daemon=false"
+      GRADLE_OPTS: "-Dorg.gradle.daemon=false -Dorg.gradle.workers.max=2"
```

Comment:
If I don't specify `worker.max`, it defaults to number of cpu. I don't want that.

Reply:
Why not? Also please add a comment for why 2 was chosen into the code